ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+        }
+      },
+      outputStrings: {
+        lightTilt: {
+          en: 'Soak dark tower',
+        },
+        darkTilt: {
+          en: 'Soak light tower',
+        },
+      },
+    },
+    // In Engravement 2 (Superchain 1), all supports or  DPS will receive lightTilt and darkTilt (2 each).
+    // All 4 also receive Heavensflame Soul.
+    // The other role group will receive lightTower, darkTower, lightBeam, and darkBeam.
+    // To resolve the Beams during the 2nd orb, lightBeam needs to stack with darkTower and both darkTilts, and vice versa.
+    // After the 3rd orb, lightTower and darkTower will drop their towers, and  darkBeam and lightBeam (respectively) will soak them.
+    // The four Heavensflame players all simultaneously need to spread to drop their AoEs.
+    // Debuffs do change based on mechanic resolution, which can complicate things:
+    // - When a lightTilt player soaks a dark beam, their debuff will change to darkTilt, and vice versa.
+    // - Once the beams detonate, the lightBeam debuff disappears and is replaced with lightTilt (same with dark).
+    // So only use the initial debuff to resolve the mechanic, and use a long suppress to avoid incorrect later alerts.
+    {
+      id: 'P12S Engravement 2 Beam Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: Object.values(engravementIdMap) },
+      condition: (data, matches) => data.engravementCounter === 2 && data.me === matches.target,
+      delaySeconds: 5, // wait until first orb resolves so as not to confuse
+      durationSeconds: 7, // keep active until beams go off
+      suppressSeconds: 30,
+      response: (_data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          lightBeam: {
+            en: 'Light Beam (stack with Dark group)',
+          },
+          darkBeam: {
+            en: 'Dark Beam (stack with Light group)',
+          },
+          lightTower: {
+            en: 'Light Tower (stack with light group)',
+          },
+          darkTower: {
+            en: 'Dark Tower (stack with dark group)',
+          },
+          lightTilt: {
+            en: 'Stack (light group) - spread later',
+          },
+          darkTilt: {
+            en: 'Stack (dark group) - spread later',
+          },
+        };
+
+        const engraveLabel = engravementLabelMap[matches.effectId];
+        if (engraveLabel === undefined)
+          return;
+        const engraveOutputStr = output[engraveLabel]!();
+        if (engraveOutputStr === undefined)
+          return;
+        if (engravementBeamIds.indexOf(matches.effectId) !== -1)
+          return { alarmText: engraveOutputStr };
+        return { alertText: engraveOutputStr };
+      },
+    },
+    // darkTower/lightTower are 20s, but lightBeam/darkBeam are shorter and swap to lightTilt/darkTilt before the mechanic resolves.
+    // So use a fixed delay rather than one based on effect duration.
```

Comment:
Do you think we should try to handle weird edge cases, like ... if both tilts get hit by towers, I think they could theoretically swap towers.  Or, if somebody dies, a spread could technically take a tower.  Or, if you do get hit by a tower drop as a tilt, then you shouldn't be told to take a tower (since it's not going to work).

Reply:
There's **soooo** many weird edge cases because of how the debuffs interact.  But just tackling those you listed:
  - If both Tilts (I think you mean the Beams that get converted to Tilts, right?) get hit by towers, I think they just die from the tower drop.  Have to go back and look at logs, but that's my recollection.  But realistically, it's such a short time in which the tower must be soaked that I don't think an alert alone would let them react quickly enough to swap unless the towers were extremely close together (which in at least the strat I've most commonly seen, they aren't).
   - If the Beams die, a Tilt (spread) could take a tower, but their Soul AoE will go off at the same time, which will kill anyone nearby (inb4 overgearing).  Would we want to alert all possible Tilt players?
    - If you're a Beam and you get hit by a tower, you shouldn't soak it, true -- but the Soak alert goes off at the same time as the Drop Tower alert (4 sec. before) - so it's not really so much that you'll be getting an inaccurate alert as that the alert you got 4 seconds ago is no longer valid because you screwed up the mechanic in the meantime.

I thought about some of these (although not as many as you!) when writing this, but where I ultimately landed was that it would be a lot of extra work to track player deaths and (unexpected) debuff changes during the mechanic and then try to write trigger contingencies for each that also don't conflict with each other.  (Couple that with the fact that I'll never get my static to willingly edge-test these, and I don't know how long PF will tolerate me griefing to test them myself.)  Also, the tower soak time is so small, I'm not sure that alert calls will help recover if people aren't already paying extremely close attention and reacting already, which makes the trigger less useful.

Reply:
Yeah, that's fair.  I feel like we've saved like one (1) pull ever with some quick reacting in this case.  Mostly just curious what if anything could be done here.  Sounds like this is all a "keep it simple for now, and reconsider later if it ever comes up".

I do think tilts can survive a tower drop.  I think they take the same damage the tower person does but their debuff is also swapped to be the same as the tower so they have to take the other tower (if any).

Reply:
I honestly can't remember, so you might be right.  But yeah, I'll add a 'todo' note so it's there for posterity.